### DISCUSSION TITLE
Renew kong api certificates snis array

### DISCUSSION THREAD

**Discussion by carlosmmatag** [Kong/kong#9841] [Help]
Hello. I am renewing the kong certificate that I use for the publication of the apis. When I first added the certificate I used this command: curl -X POST http://localhost:8001/certificates -H 'Content-Type: multipart/form-data' -F cert=@./mywildcard.crt -F key =@./mywildcard.key -F snis[]=mydomain.com and everything is fine, but now that I have to renew it, when I execute the same command but pointing to the new certificate, this appears:

{"name":"schema violation","message":"schema violation (snis: mydoamin.com already associated with existing certificate '0af45a55-143a-40ab-921f-111111111111')","fields":{"snis" :"mydomain.com already associated with existing certificate '0af45a55-143a-40ab-921f-111111111111'"},"code":2}

I have not found how to remove the snis array or unassociate it. Consequently I have not been able to add the new certificate, which is a wildcard by the way.

**Comment by hbagdi** [comment]
Please use a PUT or PATCH request to update the certificate in place. You will need the ID of the Certificate resource in Kong to perform this action. You can grab the ID by listing all the certificates (GET /certificates).


See https://docs.konghq.com/gateway/latest/admin-api/#create-or-update-certificate-associated-to-a-specific-upstream

**Comment by carlosmmatag** [comment]
Thanks man, Apreciate it. The only thing missing was the local kong url before /certificates/certificates ID and pass certificates and privatekey as a parameters. I had to figure that out. I am not very knowledgeable about this. 

In any case, for others who have problems of this type, I leave three commands. The first is for when they are going to add the certificate for the first time, the second to obtain the list of certificates with their respective ID and SNIS and the third for when it is time to renew it:

Into the certificates path:

curl -X POST http://localhost:8001/certificates/certificate ID -H 'Content-Type:multipart/form-data' -F cert=@./certificate.crt -F key=@./privatekey.key -F snis[]=subdomain.mydomain.com

curl -k -X GET http://localhost:8001/certificates

curl -X PUT http://localhost:8001/certificates/certificate ID -H 'Content-Type:multipart/form-data' -F cert=@./certificate.crt -F key=@./privatekey.key -F snis[]=subdomain.mydomain.com

thanks again.
